File: aurafit-frontend/src/Models/DTOS/CreatePromotionByProductDTO.ts
import {Product} from "../Product.ts";


export class CreatePromotionByProductDTO {
    name: string;
    discountPercent: number;
    startTime: Date;
    endTime: Date;
    product: Product;

    constructor(name: string, discountPercent: number, startTime: Date, endTime: Date, product: Product) {
        this.name = name;
        this.discountPercent = discountPercent;
        this.startTime = startTime;
        this.endTime = endTime;
        this.product = product;
    }
}

File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/promotion/CreatePromotionByProductDTO.java
package app.aurafitbackend.promotion;

import app.aurafitbackend.product.Product;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreatePromotionByProductDTO {

    private String name;
    private Integer discountPercent;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private Product product;
}
